lms/djangoapps/discussion_api/views.py

```diff
@@ -75,27 +76,41 @@ class CourseTopicsView(DeveloperErrorViewMixin, APIView):
     **Example Requests**:
 
         GET /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015
+            ?topic_id=a29d117faf43c7c59eda7717e722e9d8f4a47ea8
 
     **Response Values**:
+        if topic_id is not given:
+            * courseware_topics: The list of topic trees for courseware-linked
+              topics. Each item in the list includes:
 
-        * courseware_topics: The list of topic trees for courseware-linked
-          topics. Each item in the list includes:
+                * id: The id of the discussion topic (null for a topic that only
+                  has children but cannot contain threads itself).
 
-            * id: The id of the discussion topic (null for a topic that only
-              has children but cannot contain threads itself).
+                * name: The display name of the topic.
+
+                * children: A list of child subtrees of the same format.
+
+            * non_courseware_topics: The list of topic trees that are not linked to
+                  courseware. Items are of the same format as in courseware_topics.
+
+        if topic_id is given:
+            * id: The id of the discussion topic.
 
             * name: The display name of the topic.
 
             * children: A list of child subtrees of the same format.
-
-        * non_courseware_topics: The list of topic trees that are not linked to
-              courseware. Items are of the same format as in courseware_topics.
     """
```

Comment:
@wajeeha-khalid There was further discussion about the above in the API room on HipChat, but to summarize, I think what we'd like to see is that we use the same URL structure you've already implemented, with topic_id as a keyword parameter, but keep the same response structure as when the param is missing, so that when someone requests:

    /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015/?topic_id=a29d117faf43c7c59eda7717e722e9d8f4a47ea8

they will get a response that looks like:

    {"courseware_topics": [{<topic>}], "non_courseware_topics": []}

or

    {"courseware_topics": [], "non_courseware_topics": [{<topic>}]}

Additionally, can you make topic_id a MultiValueField, so the user can request more than one topic_id in a single request (to make it consistent with other discussion endpoints)?